### PR TITLE
Manage storage for ssl hooks

### PR DESCRIPTION
**Pull request by cmcfarlen**

### CODE REVIEW

File: src/iocore/net/SSLAPIHooks.cc [hunk 1]
```diff
@@ -22,11 +22,11 @@
...
-SSLAPIHooks *g_ssl_hooks = nullptr;
-
-void
-init_global_ssl_hooks()
+SSLAPIHooks *
+SSLAPIHooks::instance()
 {
-  g_ssl_hooks = new SSLAPIHooks;
+  static auto hooks = std::make_unique<SSLAPIHooks>();
+  return hooks.get();
```
Comment (ywkaras) [inline src/iocore/net/SSLAPIHooks.cc:31]: Why not just?
```
static SSLAPIHooks hooks;
return &hooks;
```

Comment (cmcfarlen) [reply src/iocore/net/SSLAPIHooks.cc:31]: It was created with new previously so did not want to change that, but did want to auto manage the pointer.

**Review by ywkaras — APPROVED** [review]
Uncle

**Comment by cmcfarlen** [comment]
Cherry-picked to v10.0.x

**Pull request merged**